src/twisted/newsfragments/12122.feature

```diff
@@ -0,0 +1 @@
+``twisted.internet.defer.Deferred`` operations use less CPU.
```

Comment:
Maybe something like this to highlight which methods are affected and provide a workaround.

```suggestion
twisted.internet.defer.Deferred.callback() and twisted.internet.defer.Deferred.addCallbacks() no longer use `assert` to check the type of the arguments. You should now use type checking to validate your code. These changes were done to reduce the CPU usage.
```